specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2020-05-15/timeseriesinsights.json

```diff
@@ -0,0 +1,2549 @@
+        }
+      ],
+      "description": "An environment receives data from one or more event sources. Each event source has associated connection info that allows the Time Series Insights ingress pipeline to connect to and pull data from the event source"
+    },
+    "EventHubEventSourceResource": {
+      "x-ms-discriminator-value": "Microsoft.EventHub",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/EventHubEventSourceResourceProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/EventSourceResource"
+        }
+      ],
+      "description": "An event source that receives its data from an Azure EventHub."
+    },
+    "IoTHubEventSourceResource": {
+      "x-ms-discriminator-value": "Microsoft.IoTHub",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/IoTHubEventSourceResourceProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/EventSourceResource"
+        }
+      ],
+      "description": "An event source that receives its data from an Azure IoTHub."
+    },
+    "EventSourceCommonProperties": {
+      "properties": {
+        "timestampPropertyName": {
+          "type": "string",
+          "description": "The event property that will be used as the event source's timestamp. If a value isn't specified for timestampPropertyName, or if null or empty-string is specified, the event creation time will be used."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceProperties"
+        }
+      ],
+      "description": "Properties of the event source."
+    },
+    "AzureEventSourceProperties": {
+      "properties": {
+        "eventSourceResourceId": {
+          "type": "string",
+          "description": "The resource id of the event source in Azure Resource Manager."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/EventSourceCommonProperties"
+        }
+      ],
+      "required": [
+        "eventSourceResourceId"
+      ],
+      "description": "Properties of an event source that reads events from an event broker in Azure."
+    },
+    "EventHubEventSourceCommonProperties": {
+      "properties": {
+        "serviceBusNamespace": {
+          "type": "string",
+          "description": "The name of the service bus that contains the event hub."
+        },
+        "eventHubName": {
+          "type": "string",
+          "description": "The name of the event hub."
+        },
+        "consumerGroupName": {
+          "type": "string",
+          "description": "The name of the event hub's consumer group that holds the partitions from which events will be read."
+        },
+        "keyName": {
```

Comment:
>keyName [](start = 9, length = 7)

This is not a secret, right?

Reply:
No, `sharedAccessKey` - a bit below - is. Any recommendations there?

Reply:
Yeah, you should add `"x-ms-secret": true` to any property that is a secret.

---
In reply to: [401787589](https://github.com/Azure/azure-rest-api-specs/pull/8791#discussion_r401787589) [](ancestors = 401787589)